test/dsa_no_digest_size_test.c

```diff
@@ -0,0 +1,156 @@
+/*
+ * Copyright 2018 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ * Licensed under the OpenSSL license (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+
+#include <stdlib.h>
+#include <string.h>
+
+#include "testutil.h"
+
+#include <openssl/evp.h>
+#include <openssl/dsa.h>
+#include <openssl/err.h>
+#include <openssl/rand.h>
+
+static DSA *dsa;
+
+static int genkeys(void)
+{
+    int rv;
+
+    if (!TEST_ptr(dsa = DSA_new()))
+        return 0;
+
+    rv = DSA_generate_parameters_ex(dsa, 2048, NULL, 0, NULL, NULL, NULL);
```

Comment:
While it's hardly noticeable on people's desktop computers, it can be quite a bummer on things like Raspberry Pi, and even more so under qemu. It wouldn't be inappropriate if parameters were pre-generated. I wonder if it's possible to use test/recipes/04-test_pem_data/dsaparam.pem. Alternatively one can use dsaparam -C output [which actually needs a face lift]...

Reply:
> Alternatively one can use dsaparam -C output [which actually needs a face lift]...

dsaparam -C output is fixed in #6758.